src/HomeworkDataTypes.java

```diff
@@ -0,0 +1,431 @@
+            sum += myArray[i];
+            System.out.println(myArray[i] + " ");
+        }
+        double avg = (double) (sum + 2) / a;
+        System.out.println("The average is: " + avg);
+    }
+
+    /* -------------------------------------------------------------- */
+    // 4, Write a java program that will test whether the first and the last element of an int array are the same.
+    // We can assume the array has 2 or more elements.
+    // For example:
+    // For the following input [13,6,2,8,2,3] the output should be : FALSE
+    // For the following input [13,6,2,8,2,13] the output should be : TRUE
+
+    public void testSame(int myArray[]) {
+        boolean isSame = false;
+        if (myArray[0] == myArray[myArray.length - 1]) {
+            isSame = true;
+            System.out.println("TRUE");
+        } else {
+            isSame = false;
+            System.out.println("FALSE");
+        }
+    }
+
+    /* -------------------------------------------------------------- */
+    // 5. Write a program that test whether 2 int arrays are equal.
+    // Equality in this case means that both arrays have the same number of elements an element
+    // on a specific index in first array is equal to the element on the same index in the second array.
+    // For example:
+    // For the following arrays [1,2,3,4] and [1,2,3,4] the output should be TRUE
+    // For the following arrays [] and [] the output should be TRUE
+    // For the following arrays [1,3,2] and [1,3] the output should be FALSE
+    // For the following arrays [1,2,3,4] and [1,2,4,3] the output should be FALSE
+    public void arrayEquity(int first[], int second[]) {
+        int sizeFirst = first.length;
+        int sizeSecond = second.length;
+        int counterForEquity = 0;
+        if (sizeFirst == 0 && sizeSecond == 0) {
+            System.out.println("TRUE");
+        }
+        if (sizeFirst != sizeSecond) {
+            System.out.println("FALSE");
+        } else {
+            for (int i = 0; i <= sizeFirst - 1; i++) {
+                if (first[i] == second[i]) {
+                    counterForEquity++;
+                }
+            }
+            if (counterForEquity == sizeFirst) {
+                System.out.println("TRUE");
+            } else {
+                System.out.println("FALSE");
+            }
+        }
+    }
+
+    /* -------------------------------------------------------------- */
+    // 6.Write a java program that will concatenate 2 arrays.
+    // For example
+    // Given 2 arrays [1,2,3,4] and [5,6,7,8] the output should be a new array containing [1,2,3,4,5,6,7,8]
+    public void concatenateArray(int first[], int second[]) {
+        int sizeFirst = first.length;
+        int sizeSecond = second.length;
+        int sizeNewArray = sizeFirst + sizeSecond;
+
+        int result[] = new int[sizeNewArray];
+        int position = 0;
+        for (int element : first) {
+            result[position] = element;
```

Comment:
un trick ar fi sa scrii `result[position++]` si nu mai ai nevoie de a doua linie. Asta o sa evalueze intai valoarea lui position si o sa o foloseasca in `result[position]` si apoi va incrementa valorea. 

Deci, pentru un position=3 o sa foloseasca intai valoarea 3 `position[3] si apoi o sa transforme position in 4